tests/integration/test_sca/test_scan_results.py

```diff
@@ -0,0 +1,140 @@
+    '''
+    description: This test will check that a SCA scan is correctly executed on an agent, with a given policy file and
+                 a regex engine. For this it will copy a policy file located in the data folder and verify the engine
+                 used, the amount of results found, and that the results come from the policy file.
+
+    test_phases:
+        - Copy cis_sca ruleset file into agent.
+        - Restart wazuh.
+        - Check in the log that the sca module started appears.
+        - Check the regex engine used by the policy.
+        - Get the result for each ID check
+        - Check that the policy_id from the scan matches with the file used.
+
+
+    wazuh_min_version: 4.5.0
+
+    tier: 0
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Wazuh configuration data. Needed for set_wazuh_configuration fixture.
+        - metadata:
+            type: dict
+            brief: Wazuh configuration metadata.
+        - prepare_cis_policies_file:
+            type: fixture
+            brief: copy test sca policy file. Delete it after test.
+        - set_wazuh_configuration_with_local_internal_options:
+            type: fixture
+            brief: Set the wazuh configuration and local_internal_options according to the configuration data.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - restart_modulesd_function:
+            type: fixture
+            brief: Restart the wazuh-modulesd daemon.
+        - wait_for_sca_enabled:
+            type: fixture
+            brief: Wait for the sca Module to start before starting the test.
+
+    assertions:
+        - Verify that when the `enabled` option is set to `yes`, the SCA module is enabled.
+        - Assert the engine used matches the regex_type configured in the metadata
+        - Assert the scan gets results from each rule check
+
+    input_description:
+        - The `cases_scan_results.yaml` file provides the module configuration for this test.
+        - the cis*.yaml files located in the policies folder provide the sca rules to check.
+
+    expected_output:
+        - r'.*sca.*INFO: (Module started.)'
+        - r'.*sca.*INFO: (Starting Security Configuration Assessment scan).'
+        - r".*sca.*DEBUG: SCA will use '(.*)' engine to check the rules."
+        - r".*sca.*wm_sca_hash_integrity.*DEBUG: ID: (\\d+); Result: '(.*)'"
+        - r'.*sca_send_alert.*Sending event: (.*)'
+    '''
+
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Wait for the end of SCA scan
+    evm.check_sca_scan_started(wazuh_log_monitor)
+
+    # Check the regex engine used by SCA
+    engine = evm.check_scan_regex_engine(wazuh_log_monitor)
+    assert engine == metadata['regex_type'], f"Wrong regex_engine found: expected{metadata['regex_type']}"
```

Comment:
Could you improve the message? I would like that it shows the wrong regular expression and the expected one, which might help when the test fails.

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/3566/commits/a7568fed68686051305e91f9724039ede78d0c68